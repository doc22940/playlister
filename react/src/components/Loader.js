import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class Loader extends React.Component {
  render() {
    const { isLoading } = this.props;
    const classes = classnames({
      'content-loader': true,
      'loading-modal': true,
      'justify-content-center': true,
      'display-none': !isLoading,
    });

    return (
      <div className={classes}>
        <div className='loader'></div>
      </div>
    );
  }
}

Loader.propTypes= {
  isLoading: PropTypes.bool.isRequired
};

export default Loader;
