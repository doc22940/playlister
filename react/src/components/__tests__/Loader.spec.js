import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '../Loader';

it('Loader renders correctly when isLoading=true', () => {
  const tree = renderer
    .create(<Loader isLoading />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('Loader renders correctly when isLoading=false', () => {
  const tree = renderer
    .create(<Loader isLoading={false} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

