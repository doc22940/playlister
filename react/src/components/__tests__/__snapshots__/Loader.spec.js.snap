// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly when isLoading=false 1`] = `
<div
  className="content-loader loading-modal justify-content-center display-none"
>
  <div
    className="loader"
  />
</div>
`;

exports[`Loader renders correctly when isLoading=true 1`] = `
<div
  className="content-loader loading-modal justify-content-center"
>
  <div
    className="loader"
  />
</div>
`;
